### PR TITLE
Updated to new directory structure, and to decompress lzma

### PR DESCRIPTION
**Pull request by stephanheilner**

### CODE REVIEW

File: gospellibrary/catalogs.py [hunk 1]
```diff
@@ -46,18 +49,19 @@ def exists(self):
...
             if r.status_code == 200:
                 try:
                     os.makedirs(os.path.dirname(catalog_path))
                 except OSError:
                     pass
 
-                with ZipFile(Bytes(r.content), 'r') as catalog_zip_file:
-                    catalog_zip_file.extractall(os.path.dirname(catalog_path))
+                with lzma.open(r.content) as catalog_xz_file:
+                    #TODO: save decompressed file to os.path.dirname(catalog_path)
```
Comment (hiltonc) [inline gospellibrary/catalogs.py]: I tried to test this but my local Python doesn't have `lmza` available. I think something like this should work:

```python
with lzma.open(r.content) as catalog_xz_file:
    with open(catalog_path, 'wb') as f:
        f.write(catalog_xz_file.read())
```

Comment (hiltonc) [reply gospellibrary/catalogs.py]: actually, this appears to work:

```python
with lzma.open(BytesIO(r.content)) as catalog_xz_file:
    with open(catalog_path, 'wb') as f:
        f.write(catalog_xz_file.read())
```

File: gospellibrary/catalogs.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import requests
 import os
-from zipfile import ZipFile
+import lzma
```
Comment (hiltonc) [inline gospellibrary/catalogs.py]: For when you have everything working, let's do this to support Python < 3.3, or for when lzma wasn't installed on the system:

```python
try:
    import lzma
except ImportError:
    from backports import lzma
```

Be sure to also:
1. `pip install backports.lzma`
2. `pip freeze > requirements.txt`,
3. Add it to `setup.py`

**Comment by stephanheilner** [comment]
I don't have the ability to test this locally, do you?

**Comment by hiltonc** [comment]
This works, but there are more issues because of changes in the database format:

```
$ python
>>> from gospellibrary.catalogs import CatalogDB
>>> from gospellibrary.item_packages import ItemPackage
>>> catalog = CatalogDB()
>>> item = catalog.item(uri='/scriptures/bofm', lang='eng')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/cahilton/Documents/Code/python-gospel-library/gospellibrary/catalogs.py", line 232, in item
    '''SELECT item.* FROM item INNER JOIN language ON item.language_id=language._id WHERE uri=? AND iso639_3=?''', [uri, lang])
sqlite3.OperationalError: no such column: language._id
```

**Comment by stephanheilner** [comment]
I can't figure out how to run this locally to fix the rest of the issues. :(

**Comment by hiltonc** [comment]
I'm not sure when I'll have time but I'll try to look into it.

**Comment by hiltonc** [comment]
I've updated `master` with support for v4. There are a few breaking changes:

* `CatalogDB().item(uri=item_uri, lang='eng')` changes to `CatalogDB(iso639_3_code='eng').item(uri=item_uri)`
* `ItemPackage(item_external_id=item['external_id'], item_version=item['version'])` changes to `ItemPackage(iso639_3_code='eng', item_id=item['id'], item_version=item['version'])`

Let me know if you run into any issues with it.

**Comment by stephanheilner** [comment]
Awesome!!

**Pull request closed without merging**